src/vscode-dts/vscode.d.ts

```diff
@@ -1312,6 +1312,10 @@ declare module 'vscode' {
 			 * Add undo stop after making the edits.
 			 */
 			readonly undoStopAfter: boolean;
+			/**
+			 * Adjust the indentation of the snippet.
+			 */
+			readonly adjustWhitespace?: boolean;
```

Comment:
In the API optional properties default to `false` and that's not the case here and also undesirable (we cannot change the existing behaviour here). We would need a name that's the other way around, I suggest `keepWhitespace` which is already used for `CompletionItem` for the same/similar purpose

Reply:
to double-click. Only change the API name and do the conversion in the API implementation [here](https://github.com/microsoft/vscode/blob/0c7bdaaae9b654d97ee5b212bbf4b81bd871540f/src/vs/workbench/api/common/extHostTextEditor.ts#L524)

Reply:
I have addressed this (I think). I didn't know `extHostTextEditor` ends up calling `mainThreadEditor` but that makes more sense now. So `keepWhitespace` is optional, and when undefined, we set it to `false` (current default behaviour).